src/wp-admin/includes/image.php

```diff
@@ -331,18 +364,73 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 
 			if ( true === $rotated ) {
 				// Append `-rotated` to the image file name.
-				$saved = $editor->save( $editor->generate_filename( 'rotated' ) );
+				$saved = $editor->save( $editor->generate_filename( 'rotated' ), $primary_mime_type );
 
 				if ( ! is_wp_error( $saved ) ) {
 					$image_meta = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
 
+					$image_meta['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $saved );
+
 					// Update the stored EXIF data.
 					if ( ! empty( $image_meta['image_meta']['orientation'] ) ) {
 						$image_meta['image_meta']['orientation'] = 1;
 					}
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
 				} else {
 					// TODO: Log errors.
 				}
+
+				// Save additional mime types.
+				foreach ( $additional_mime_types as $output_mime_type ) {
+					$editor = wp_get_image_editor( $file );
+
+					if ( is_wp_error( $editor ) ) {
+						// This image cannot be edited.
+						continue;
+					}
+
+					if ( empty( $mime_extension_map[ $output_mime_type ] ) ) {
+						// Skip mime types that are not supported.
+						continue;
+					}
+
+					$saved = $editor->save( $editor->generate_filename( 'rotated', null, $mime_extension_map[ $output_mime_type ] ), $output_mime_type );
+					if ( ! is_wp_error( $saved ) ) {
+						$image_meta['sources'][ $output_mime_type ] = _wp_get_sources_from_meta( $saved );
+						wp_update_attachment_metadata( $attachment_id, $image_meta );
+					}
+				}
+			}
+		}
+		// Ensure presence a top level sources array for the full size images.
+		if ( ! isset( $image_meta['sources'] ) || ! is_array( $image_meta['sources'] ) ) {
+			$image_meta['sources'] = array();
+		}
+
+		// Populate the top level primary mime type data.
+		if ( empty( $image_meta['sources'][ $primary_mime_type ] ) ) {
+			$image_meta['sources'][ $primary_mime_type ] = array(
+				'file'     => wp_basename( $file ),
+				'filesize' => isset( $image_meta['filesize'] ) ? $image_meta['filesize'] : wp_filesize( $file ),
+			);
+			wp_update_attachment_metadata( $attachment_id, $image_meta );
+		}
+
+		$editor = wp_get_image_editor( $file );
+
+		if ( is_wp_error( $editor ) ) {
+			// This image cannot be edited.
+			return $image_meta;
+		}
+
+		// Populate the top level additional mime type data.
+		foreach ( $additional_mime_types as $output_mime_type ) {
+			if ( empty( $image_meta['sources'][ $output_mime_type ] ) ) {
+				$saved = $editor->save( $editor->generate_filename( str_replace( 'image/', '', $output_mime_type ) ), $output_mime_type );
+				if ( ! is_wp_error( $saved ) ) {
+					$image_meta['sources'][ $output_mime_type ] = _wp_get_sources_from_meta( $saved );
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
+				}
 			}
 		}
```

Comment:
What is the purpose of this code block? It doesn't make sense to me, also it still tries to guess the extension from the MIME type name which is not correct.

Is this an accidental leftover from previous code?

Reply:
I believe this is required to generate the additional full sized (not -scaled or -rotated) images, I'll fix the extension issue, that is old code :) I'll see if I can move this above -scaled/rotated code so we don't need to re-initialize the editor

Reply:
removed editor reinitialization here